Orange/widgets/visualize/owvenndiagram.py

```diff
@@ -649,21 +646,21 @@ def commit(self):
             set.union, [self.disjoint[index] for index in self.selection],
             set()
         )
-        selected_keys = reduce(set.union,
-                               [set(self.area_keys[area]) for area in self.selection],
-                               set())
+        selected_keys = reduce(
+            set.union, [set(self.area_keys[area]) for area in self.selection],
+            set())
         selected = None
 
         if self.rowwise:
             selected_ids = self.get_indices_to_match_by(selected_keys, self.selection)
-            annotated_ids = self.get_indices_to_match_by(self.data.keys())
-            annotated = self.create_from_rows(self.data.keys(), annotated_ids, True)
+            annotated_ids = self.get_indices_to_match_by(self.data)
+            annotated = self.create_from_rows(self.data, annotated_ids, True)
             if self.selected_items:
                 selected = self.create_from_rows(selected_keys, selected_ids, False)
         else:
-            annotated = self.create_from_columns(list(self.selected_items), self.data.keys(), False)
+            annotated = self.create_from_columns(self.selected_items, self.data, False)
```

Comment:
Please check this, but as I see, `self.selected_items` is a set and `create_from_columns` checks whether it contains certain elements. If I'm correct, casting to list just hurts performance.